guardian/cdk
# check age of aws-cdk/core during ci

## What does this change?


We introduced dependabot in #55 to keep our dependencies up to date. However dependabot doesn't report against the CDK libraries (related https://github.com/dependabot/dependabot-core/issues/1171).

This adds a fairly gnarly bash script to:
- List all versions of `@aws-cdk/core` (and assume it's ordered!)
- Use `jq` to find how out of date we are
- If we are 5* versions out of date, exit 1

This is all run in CI.

Also takes inspiration [from Source](https://theguardian.design/2a1e5182b/p/876251-getting-started/t/06dcf1) and uses [`npm-check-updates`](https://www.npmjs.com/package/npm-check-updates) to update the `@aws-cdk/*` libraries. All wrapped in a new `update-aws-cdk` script.

Related to https://github.com/guardian/cdk/pull/191.

*this was an arbitrary choice, we can tweak as needed

## Does this change require changes to existing projects or CDK CLI?


Assuming #191 goes in first, then no.

## How to test


Run the two new scripts locally:
- `./script/check-aws-cdk`
- `./script/update-aws-cdk`

## How can we measure success?


We keep up to date with `@aws-cdk/*` more frequently, especially as they [release fairly often](https://github.com/aws/aws-cdk/releases).

## Have we considered potential risks?


Too much of a chore to keep up to date?
